platforms/software/dependency-management/src/main/java/org/gradle/api/internal/artifacts/configurations/DefaultConfigurationContainer.java

```diff
@@ -172,44 +175,33 @@ protected UnknownDomainObjectException createNotFoundException(String name) {
         return new UnknownConfigurationException(String.format("Configuration with name '%s' not found.", name));
     }
 
+    private RuntimeException failOnAttemptToAdd(String behavior) {
```

Comment:
I know we didn't before, but we should have a test now that verifies this behavior

Reply:
Good idea, I'll add one.